File: valid_anagram_242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        countS, countT = {}, {}

        for i in range(len(s)):
            countS[s[i]] = 1 + countS.get(s[i], 0)
            countT[t[i]] = 1 + countT.get(t[i], 0)

        return countS == countT


if __name__ == "__main__":
    s = "anagram"
    t = "nagaram"

    solution = Solution()
    result = solution.isAnagram(s, t)
    print(result)

File: Java/valid_anagram_242.java
package Java;

class ValidAnagram {
  public boolean isAnagram(String s, String t) {
    if (s.length() != t.length()) {
      return false;
    }

    int[] store = new int[26];

    for (int i = 0; i < s.length(); i++) {
      store[s.charAt(i) - 'a']++;
      store[t.charAt(i) - 'a']--;
    }

    for (int n : store) {
      if (n != 0)
        return false;
    }

    return true;
  }

  public static void main(String[] args) {
    String s = "anagram";
    String t = "nagaram";

    ValidAnagram solution = new ValidAnagram();
    boolean isAnagram = solution.isAnagram(s, t);

    System.out.println(isAnagram);
  }
}